kolibri/plugins/coach/assets/src/modules/examsRoot/handlers.js

```diff
@@ -8,6 +8,7 @@ export function showExamsPage(store, classId) {
   store.commit('SET_PAGE_NAME', PageNames.EXAMS);
 
   const promises = [
+    store.dispatch('initClassInfo', classId),
```

Comment:
@ShivangRawat30 Placing the promise here doesn't ensure it is resolved before other promises (even though it may look so randomly)

An example:
```
const promise1 = new Promise(resolve => {
  setTimeout(() => {
    console.log("Promise 1 resolved");
    resolve();
  }, 2000)
})
  
const promise2 = new Promise(resolve => {
  setTimeout(() => {
    console.log("Promise 2 resolved");
    resolve();
  }, 500)
})

Promise.all([promise1, promise2])
```

Here you can see that the second promise is actually resolved before the first one.

We need to ensure the following:

- `'initClassInfo'` and `'getFacilities'` are run first (their order doesn't matter)
- after the two above are resolved, then the remaining promises are run

You will need to utilize `.then()` or `async/await`